crates/oci-distribution/src/lib.rs

```diff
@@ -252,4 +289,24 @@ mod test {
         assert_eq!(manifest.schema_version, 2);
         assert!(!manifest.layers.is_empty());
     }
+
+    #[tokio::test]
+    async fn test_pull_layer() {
+        let image = Reference::try_from(HELLO_IMAGE).expect("parsed reference");
+        let mut c = Client::default();
+        c.auth(&image, None).await.expect("authenticated");
+        let manifest = c.pull_manifest(&image).await.expect("pull manifest");
+
+        // Pull one specific layer
+        let file: Vec<u8> = Vec::new();
+        let layer0 = manifest.layers[0].clone();
+
+        let file = c
+            .pull_layer(&image, layer0.digest.as_str(), file)
```

Comment:
nit: `as_str` isn't necessary. You can just pass the digest `String` by reference and it will automatically be coerced into a `&str`.

Reply:
Fixed